### ISSUE TITLE
NonEmptyVector unsafe apply method

### ISSUE THREAD

**Issue by peterneyens** [typelevel/cats#1214]
With the `apply` method we get because `NonEmptyVector` is a case class, we can create an empty `NonEmptyVector`.

```scala
scala> NonEmptyVector(Vector.empty[Int])
res1: cats.data.NonEmptyVector[Int] = NonEmptyVector(Vector())

scala> res1.head
java.lang.UnsupportedOperationException: empty.head
  at scala.collection.immutable.Vector.head(Vector.scala:193)
  at cats.data.NonEmptyVector.head(NonEmptyVector.scala:31)
  ... 42 elided
```

Should we turn it into a regular class with the following private `apply` method ?

```scala
private def apply[A](vector: Vector[A]): NonEmptyVector[A] =
  new NonEmptyVector(vector)
```

**Comment by ceedubs** [comment]
Ouch, good catch. For some reason I was thinking that the generated `apply` was private when a case class constructor is marked private.

> Should we turn it into a regular class

This is beginning to seem like the right solution to me. If it stays a case class we'll need to override both the `apply` and the `copy` method to make them `private`. And even then we don't _really_ want either of them to exist, because we'd prefer to just have `fromVectorUnsafe`.

We'll need to make sure that we do the right thing for `equals`, `toString`, and `hashCode` (where the easiest thing to do is probably to delegate through to the underlying `Vector` for each, and probably to prepend `NonEmpty` to the `toString`).

> with the following private apply method?

As mentioned above, I think we might just want `fromVectorUnsafe` instead.

**Comment by mikejcurry** [comment]
I believe this is resolved by #1212
